### DISCUSSION TITLE
`watch` return old value when field value set to undefined

### DISCUSSION THREAD

**Discussion by avxit** [react-hook-form/react-hook-form#7435] [Q&A]
I don't know whether it's a bug or `rhf` intended to do.

Here is the sandbox I created: https://codesandbox.io/s/mystifying-robinson-bftk9.

To reproduce:
- In step 1 hit `Next` button
- In step 2 input some value then hit `Prev` button (*)
- In step 1 hit `Next` button
- In step 2 input some value, then delete all, you'll see the value of `wi2` is `undefined`, then hit `Prev`
- In step 1 hit `Next` button, you will see old value (input at * above)

**Comment by bluebill1049** [comment]
`const form = useForm();` you should consider to provide a `defaultValues`.

**Reply by gormonn** [reply]
In other case i can't provide defaultValues for generative fields...
So they defaultChecked doesn't take effect to watch... (((
